asv_benchmarks/benchmarks/comparisons/image.py

```diff
@@ -0,0 +1,71 @@
+"""Benchmarks for image operations under nilearn.image module."""
+
+# ruff: noqa: RUF012
+
+from nilearn.image import mean_img
+
+from ..common import Benchmark
+from ..utils import load
+
+
+class CompareLoad(Benchmark):
+    """
+    A benchmark that compares the performance of loading images from
+    disk using nibabel and nilearn.
+    """
+
+    # compare loading images using nibabel and nilearn
+    param_names = ["loader"]
+    params = ["nilearn", "nibabel (ref)"]
+
+    def time_compare_load(self, loader):
+        """Time the loading of images."""
+        load(loader)
+
+    def peakmem_compare_load(self, loader):
+        """Peak memory of loading images."""
+        load(loader)
+
+
+class CompareMean(Benchmark):
+    """
+    A benchmark that compares the performance of first computing the mean
+    of a 4D image loaded via nibabel and nilearn.
+    """
+
+    # compare loading images using nibabel and nilearn
+    param_names = ["loader"]
+    params = ["nilearn", "nibabel (ref)"]
+
+    def time_compare_mean(self, loader):
+        """Time the loading followed by taking the mean."""
+        img = load(loader)[1]
+        mean_img(img)
+
+    def peakmem_compare_mean(self, loader):
+        """Peak memory of loading followed by taking the mean."""
+        img = load(loader)[1]
+        mean_img(img)
+
+
+class CompareSlice(Benchmark):
```

Comment:
I think we are pretty much stuck with the hovering thing on the webpages.
However, you could look at the code here if needed...

Reply:
Actually, all the code excerpts are there in a JSON file.
It is definitely possible to write a script to add more views/plots etc. on the benchmark webpages. 
But I would keep that as a long-term-nice-to-have.